src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+
+    private static final int DEFAULT_SIZE_OF_ARRAY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INCREASING_T = 2;
+    private Node<K, V> [] table = new Node[DEFAULT_SIZE_OF_ARRAY];
+    private Node<K, V> node;
+    private int size;
+    private int threshold;
+
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> exsitNode = getNode(key);
+        if (exsitNode != null) {
+            exsitNode.value = value;
+            return;
+        }
+        checkSize();
+        int index = getIndexOfArray(key);
```

Comment:
You call `getIndexOfArray(key)` inside `getNode(key);`
It means you call `getIndexOfArray(key)` **two times** inside put(K key, V value) - do not do it.